### PR TITLE
Avoid re-reading parquet footer for delta stats in writer

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description
Currently we're reading parquet file footer back just after
writing it to file system for computing delta stats.
Since Trino parquet writer already has the file footer, there
is no need to perform an extra IO to get the footer again.

## Additional context and related issues
Similar to https://github.com/trinodb/trino/pull/19090 but for delta

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Delta
* Improve performance of writing parquet files. ({issue}`19122`)
```

### CODE REVIEW

**Review by sopel39 — APPROVED** [review]

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java [hunk 1]
```diff
@@ -99,6 +101,23 @@ protected DistributedQueryRunner createQueryRunner()
         }
     }
 
+    @Test
+    public void testCreateTableAsSelect()
+    {
+        assertFileSystemAccesses(
+                "CREATE TABLE test_create_as_select AS SELECT 1 col_name",
+                ImmutableMultiset.<FileOperation>builder()
+                        .addCopies(new FileOperation(STARBURST_EXTENDED_STATS_JSON, "extendeded_stats.json", INPUT_FILE_NEW_STREAM), 1)
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java:110]: pre-existing `extendeded_stats.json` is gramatically incorrect. 
In case it would get changed, this would likely be in a different PR

Comment (raunaqmorarka) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java:110]: It's not changing in this PR, it's still the same spelling as before

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java [hunk 2]
```diff
@@ -196,31 +189,27 @@ public DataFileInfo getDataFileInfo()
                 creationTime,
                 dataFileType,
                 partitionValues,
-                readStatistics(inputFile, dataColumnTypes, rowCount));
+                readStatistics(fileWriter, rootTableLocation.appendPath(relativeFilePath), dataColumnTypes, rowCount));
     }
 
-    private static DeltaLakeJsonFileStatistics readStatistics(TrinoInputFile inputFile, Map</* lowercase */ String, Type> typeForColumn, long rowCount)
+    private static DeltaLakeJsonFileStatistics readStatistics(ParquetFileWriter fileWriter, Location path, Map</* lowercase */ String, Type> typeForColumn, long rowCount)
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java:195]: Instead of passing `fileWriter` , please pass `FileMetaData fileMetaData` (`fileWriter.getFileMetadata()` ) to the method.
We do _read_Statistics in this method.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java (hunk 1 above)
Comment (homar) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java:110]: it is has been here for a while and i doubt we can fix it now :D

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java [hunk 3]
```diff
@@ -99,6 +101,23 @@ protected DistributedQueryRunner createQueryRunner()
         }
     }
 
+    @Test
+    public void testCreateTableAsSelect()
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java:105]: Could you add a test for a CTAS on a partitioned table ?
I'm guessing that this test would be a bit more revealing.

**Review by findinpath — APPROVED** [review]

**Review by findinpath — APPROVED** [review]

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java [hunk 4]
```diff
@@ -99,6 +101,35 @@ protected DistributedQueryRunner createQueryRunner()
...
+    {
+        assertFileSystemAccesses(
+                "CREATE TABLE test_create_as_select AS SELECT 1 col_name",
+                ImmutableMultiset.<FileOperation>builder()
+                        .addCopies(new FileOperation(STARBURST_EXTENDED_STATS_JSON, "extendeded_stats.json", INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(STARBURST_EXTENDED_STATS_JSON, "extendeded_stats.json", INPUT_FILE_EXISTS), 1)
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java:111]: In accordance with https://github.com/trinodb/trino/pull/19128/commits/80a3ffe74c8e848a0ae5451857334ddd4f6cbec0 , you could use `add(...)` instead of `addCopies(...)`

**Review by findinpath — APPROVED** [review]
